Source/WebCore/platform/graphics/FontCascade.cpp

```diff
@@ -476,7 +476,7 @@ bool FontCascade::fastAverageCharWidthIfAvailable(float& width) const
 {
     bool success = hasValidAverageCharWidth();
     if (success)
-        width = roundf(primaryFont().avgCharWidth()); // FIXME: primaryFont() might not correspond to firstFamily().
+        width = std::max<float>(primaryFont().avgCharWidth(), roundf(primaryFont().avgCharWidth())); // FIXME: primaryFont() might not correspond to firstFamily().
```

Comment:
This operation is quite frankly kinda weird, because for values with < 0.5 decimal it returns fractional values, but for values >= 0.5 decimal it returns an integer. If we need this for compatibility, it's probably helpful to copy over the comments from https://chromium.googlesource.com/chromium/src/+/a00094b9b640c10fff554af6dcb4a68d78300321%5E%21/#F0

At least, that's my opinion. An official reviewer would have to confirm if they agree. :)

It's also a bit long for one line so you might consider

```
width = primaryFont().avgCharWidth(); // FIXME: ...
width = std::max<float>(width, roundf(width));
```

which also makes it easier to see the logic inside the max function.

Reply:
(Also if this weird thing is the required Web-compat behavior for textarea and input, we should get it specced.)

Reply:
I agree with Elika, this is a very odd looking operation. 
I also don't like the idea of doing any kind of value adjustment at the font level to support higher level concepts like in this case enclosing intrinsic width operations (other clients may also find it completely unexpected)
This (or a saner version of it) should happen in RenderTextControl where we actually have enough context to decide on the type of rounding operation (and the precision level) -which in case of intrinsic width computation is usually a simple ceil. 
(also, though at this point it's not really relevant since both arguments to std::max are float types, we don't need to say std::max<float>.